Bottom: 8e1e555675ee4f1ed8e6a0ecaf14dcd53dfc3b04
Top:    2af77d22bdea44f7c9674ec722530d8ab6de44c2
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-09 18:16:34 +0000

Refresh of flags-c-avoid-compiler-warning

---

diff --git a/src/flags.c b/src/flags.c
index fdaef44..499e59e 100644
--- a/src/flags.c
+++ b/src/flags.c
@@ -163,6 +163,11 @@ FlagLayerActive (int n)
   return current_layer == n;
 }
 
+/* The cast to (int) is ONLY valid because we know we are
+ * taking offsets on structures where the offset will fit
+ * in an integer variable. It silences compile warnings on
+ * 64bit machines.
+ */
 #define OffsetOf(a,b) (int)(size_t)(&(((a *)0)->b))
 
 HID_Flag flags_flag_list[] = {
